src/main/java/com/pusher/client/channel/impl/ChannelImpl.java

```diff
@@ -69,10 +72,12 @@ public void unbind(final String eventName, final SubscriptionEventListener liste
         validateArguments(eventName, listener);
 
         final Set<SubscriptionEventListener> listeners = eventNameToListenerMap.get(eventName);
```

Comment:
As above, you must perform both the read and the write in the critical section, otherwise there's no guarantee of consistency.